### PR TITLE
fix: adding logger for undefined source event

### PR DESCRIPTION
**Pull request by shrouti1507**
## What are the changes introduced in this PR?

We have got bugsnag error :
Screenshot 2024-11-15 at 12 05 03 PM

Adding logger to get more info
## What is the related Linear task?

Resolves INT-2873

## Please explain the objectives of your changes below

Put down any required details on the broader aspect of your changes. If there are any dependent changes, **mandatorily** mention them here

### Any changes to existing capabilities/behaviour, mention the reason & what are the changes ?

N/A

### Any new dependencies introduced with this change?

N/A

### Any new generic utility introduced or modified. Please explain the changes.

N/A

### Any technical or performance related pointers to consider with the change?

N/A

@coderabbitai review

<hr>

### Developer checklist

- [ ] My code follows the style guidelines of this project

- [ ] **No breaking changes are being introduced.**

- [ ] All related docs linked with the PR?

- [ ] All changes manually tested?

- [ ] Any documentation changes needed with this change?

- [ ] Is the PR limited to 10 file changes?

- [ ] Is the PR limited to one linear task?

- [ ] Are relevant unit and component test-cases added in **new readability format**?

### Reviewer checklist

- [ ] Is the type of change in the PR title appropriate as per the changes?

- [ ] Verified that there are no credentials or confidential data exposed with the changes.

### CODE REVIEW

**Comment by devops-github-rudderstack** [comment]
Test report for this run is available at: https://test-integrations-dev.s3.amazonaws.com/integrations-test-reports/rudder-transformer/3879/test-report.html

**Review by sanpj2292 — APPROVED** [review]

File: src/controllers/source.ts [hunk 1]
```diff
@@ -28,6 +28,11 @@ export class SourceController {
       );
       ctx.body = resplist;
     } catch (err: any) {
```
Comment (krishna2020) [inline src/controllers/source.ts:30]: No need to mention module again in error message - we have structured logging now. Focus on mentioning the error message.
`events` can't be logged, it's a security risk - how did you overlooked it ? Is there specific reason to log events ?
cc: @sanpj2292

Comment (sanpj2292) [reply src/controllers/source.ts:30]: > how did you overlooked it ? Is there specific reason to log events ?

The inclusion of `events` was a call we took to understand what is the event which is causing a problem. The reason is we don't have a lot of context on what kind of event was received to us & even though we get the error message, we would not be in a position to properly identify the problem and come up with a fix.

In this particular bugsnag issue, we saw that `sourceEvents` came in as a non-array which means there is some problem in adaption logic. And is it undefined, string, object -- we absolutely have no-clue

We will remove `events` from getting logged but we should definitely find an alternate (if not already in consideration or present) to address this

Comment (sanpj2292) [reply src/controllers/source.ts:30]: @shrouti1507 let's remove `events` from the log getting emitted for an error

File: src/controllers/source.ts [hunk 2]
```diff
@@ -4,13 +4,27 @@ import { MiscService } from '../services/misc';
...
+      version,
+    };
+  }
+
   public static async sourceTransform(ctx: Context) {
     logger.debug('Native(Source-Transform):: Request to transformer::', ctx.request.body);
     const requestMetadata = MiscService.getRequestMetadata(ctx);
     const events = ctx.request.body as object[];
     const { version, source }: { version: string; source: string } = ctx.params;
+    const enrichedMetadata = this.enrichMetadata(requestMetadata, source, version);
```
Comment (krishna2020) [inline src/controllers/source.ts:27]: do we really need a function here ?
```suggestion
    const enrichedMetadata = {
      ...metadata,
      source,
      version,
    };
```

Comment (sanpj2292) [reply src/controllers/source.ts]: Done

**Review by krishna2020 — APPROVED** [review]

File: src/controllers/source.ts
Comment (krishna2020) [inline src/controllers/source.ts:1]: If required, sourceId can also be logged from `events[0].config`

**Pull request merged**